sql/mysql/src/TMySQLStatement.cxx

```diff
@@ -313,29 +301,35 @@ void TMySQLStatement::SetBuffersNumber(Int_t numpars)
    memset(fBind, 0, sizeof(MYSQL_BIND)*fNumBuffers);
 
    fBuffer = new TParamData[fNumBuffers];
-   memset(fBuffer, 0, sizeof(TParamData)*fNumBuffers);
+   for (int n=0;n<fNumBuffers;++n) {
+      fBuffer[n].fMem = nullptr;
+      fBuffer[n].fSize = 0;
+      fBuffer[n].fSqlType = 0;
+      fBuffer[n].fSign = kFALSE;
+      fBuffer[n].fResLength = 0;
+      fBuffer[n].fResNull = false;
+      fBuffer[n].fStrBuffer.clear();
+      fBuffer[n].fFieldName.clear();
+   }
 }
 
 ////////////////////////////////////////////////////////////////////////////////
 /// Convert field value to string.
 
-const char* TMySQLStatement::ConvertToString(Int_t npar)
+const char *TMySQLStatement::ConvertToString(Int_t npar)
 {
-   if (fBuffer[npar].fResNull) return 0;
+   if (fBuffer[npar].fResNull) return nullptr;
 
-   void* addr = fBuffer[npar].fMem;
+   void *addr = fBuffer[npar].fMem;
    Bool_t sig = fBuffer[npar].fSign;
 
-   if (addr==0) return 0;
+   if (!addr) return nullptr;
 
    if ((fBind[npar].buffer_type==MYSQL_TYPE_STRING) ||
       (fBind[npar].buffer_type==MYSQL_TYPE_VAR_STRING))
-      return (const char*) addr;
-
-   if (fBuffer[npar].fStrBuffer==0)
-      fBuffer[npar].fStrBuffer = new char[100];
+      return (const char *) addr;
 
-   char* buf = fBuffer[npar].fStrBuffer;
+   char buf[100];
```

Comment:
What 'guarantees' that 100 is enough?

Reply:
Kind of reasonable size. 
Here basic data types like int or date are converted.

Reply:
Can you add something like:
```
if (npar > 100) {
   Error( .... )
   return error_code;
}
```
?

Reply:
No, it is not about of parameter number. It is local char buffer for printf. Any number of parameters in SQL statement still allowed

Reply:
I see.  I did not read the code below.  

Could we replace 100 with a constant (used in all the lines below)?

Could the return value of snprintf (below) be checked (and the error case handled)?

Reply:
> Could the return value of snprintf (below) be checked (and the error case handled)?

Done

Reply:
Did you upload the related commits?

Reply:
Yes, this one: https://github.com/root-project/root/pull/4927/commits/561e7cc43ecc3a30e80d0883d7a3301edc2138d3

Reply:
Thanks (github was 'hiding' it from me :( )